Exercise_5.java

```diff
@@ -2,19 +2,44 @@
     void swap(int arr[], int i, int j) 
     { 
 	//Try swapping without extra variable 
+    arr[i] = arr[i] + arr[j];
+    arr[j] = arr[i] - arr[j];
+    arr[i] = arr[i] - arr[j];
     } 
   
     /* This function is same in both iterative and 
        recursive*/
     int partition(int arr[], int l, int h) 
     { 
         //Compare elements and swap.
+        int low = l;
+        int pivot = h;
+        for(int i = l; i<=h; i++) {
+            if(arr[i] < arr[pivot]) {
+                swap(arr,i,low);
+            }
+        }
+        swap(arr,low,pivot);
+        return low;
     } 
   
     // Sorts arr[l..h] using iterative QuickSort 
     void QuickSort(int arr[], int l, int h) 
     { 
         //Try using Stack Data Structure to remove recursion.
+        int[] stack = new int[h-l+1];
+        int top = -1;
+        stack[++top] = l;
+        stack[++top] = h;
+        while(top >= 0) {
+            h = stack[top--];
+            l = stack[top--];
+            int partitionPosition = partition(arr,l,h);
+    

```

Comment:
You were proceeding in the right direction, from here compare L & H with partitionPosition and place elements accordingly. You can text me on Slack for any further doubts.